src/t/d17.clj

```diff
@@ -0,0 +1,258 @@
+(ns t.d17
+  (:gen-class)
+  (:require
+    [clojure.math :as math]
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]
+    [clojure.core.match :refer [match]]))
+
+(def sample
+"Register A: 729
+Register B: 0
+Register C: 0
+
+Program: 0,1,5,4,3,0")
+(def sol "4,6,3,5,6,3,5,2,1,0")
+
+(def sample2
+"Register A: 2024
+Register B: 0
+Register C: 0
+
+Program: 0,3,5,4,3,0")
+(def sol2 117440)
+
+(defn parse-input
+  [input]
+  (let [ [_ A B C instr] (re-find #"Register A: (\d+)\nRegister B: (\d+)\nRegister C: (\d+)\n\nProgram: ([\d,?]+)" input)
+        instr (->> (cljstr/split instr #",")
+                   (map parse-long)
+                   vec
+                   )]
+    [(parse-long A) (parse-long B) (parse-long C) instr]))
+
+
+(defn combo-oprand
+  [A B C oprand]
+  (cond
+    (<= oprand 3) oprand
+    (= 4 oprand) A
+    (= 5 oprand) B
+    (= 6 oprand) C
+    "7 or none of the above is pbm" (prn "Problem with oprand" oprand)))
```

Comment:
Nice! Possibly:
```suggestion
    "7 or none of the above is pbm" (throw (ex-info "Problem with oprand" {:combo oprand}))))
```
if you want your program to stop with an exception here rather than this function returning `nil` (the value of `(prn ...)`) and blowing up later when, say, trying to add `nil` to something.